### DISCUSSION TITLE
How to load plugin that loads existing plugins? Adding new extras.lang

### DISCUSSION THREAD

**Discussion by lbennett-stacki** [LazyVim/LazyVim#3122] [Q&A] [answered]
I've been looking at docs for loading local plugins but can't figure it out.

I'm adding my own local swift lang lazy plugin with intention to add it later to official `extras.lang`.

I start with the root lazy config plugin import:

```lua
{
	"swift",
	dir = "/Users/.../workspace/.../extras/lang/swift",
	name = "swift",
},
```

And then in `/Users/.../workspace/.../extras/lang/swift` I have an `init.lua`:

```lua
return {
	{
		"nvim-treesitter/nvim-treesitter",
		opts = function(_, opts)
			if type(opts.ensure_installed) == "table" then
				vim.list_extend(opts.ensure_installed, {
					"swift",
				})
			end
		end,
	},
}
```

I can see it in the plugin list when I `:Lazy` but it has an orange dot and just shows the dir path. If I check treesitter health I see no swift installed, so it must not have consumed the plugin as I expected. I can install the tree sitter grammar manually with tree-sitter commands so it does exist.

**Comment by dpetka2001** [comment, marked as answer]
Just fork the repo and clone it locally and do whatever changes you want in another branch in the directory of extras.lang

**Reply by lbennett-stacki** [reply]
Ah this does sound smart, I'll give it a go

**Reply by dpetka2001** [reply]
Fork the repo and push your changes to your own fork instead of creating draft PRs here. After you're finished with your PR in your fork then make a PR here. Now you just added a draft PR here, which isn't really ideal.

**Reply by lbennett-stacki** [reply]
Thanks @dpetka2001 this is progressing much faster now. I had a thought, with swift, if people want to use it for an xcode project, there is an extra step they must take, they must install a binary [xcode-build-server](https://github.com/SolaWing/xcode-build-server) and run it's `config` command in their project root.

Is there a recommended way to do this or should I focus on documentation for that?

**Reply by lbennett-stacki** [reply]
Another example is a linter binary https://github.com/realm/SwiftLint

**Reply by dpetka2001** [reply]
You could take a look at Mason and ask its maintainer if he's willing to incorporate these binaries into Mason itself (if it's possible). That would make their distribution a lot easier. If he's not willing to do so, then the users would have to do so themselves and set it up correctly. 

Otherwise, you would just have to focus on a proper documentation about how users can install those tools themselves.

**Reply by lbennett-stacki** [reply]
Great, thanks again!
